### ISSUE TITLE
Show sign in page after user log out when using a single oauth2 provider

### ISSUE THREAD

**Issue by shawnzhu** [dpgaspar/Flask-AppBuilder#1705]
If you'd like to report a bug in Flask-Appbuilder, fill out the template below. Provide
any extra information that may be useful

Responsible disclosure:
We want to keep Flask-AppBuilder safe for everyone. If you've discovered a security vulnerability
please report to danielvazgaspar@gmail.com.

### Environment

Flask-Appbuilder version: 3.3.3

pip freeze output:

### Describe the expected results

Given a **single** OAuth2 provider configured, when a signed in user tries to log out, it should log out user.

https://github.com/dpgaspar/Flask-AppBuilder/blob/211284bc5792d2a24a486f5c2b0f4fc86de89fbe/flask_appbuilder/security/views.py#L607-L615

### Describe the actual results

User will be automatically signed in after trying logging out. 

```pytb
Paste the full traceback if there was an exception.
```

### Steps to reproduce

When a sigined in over the configured OAuth2 provider, after visit `/logout`, it will redirect user back to `/login` then sign in user automatically since the logout flow really doesn't log user out of OAuth2 provider.

**Comment by dpgaspar** [comment]
Hi @shawnzhu thank for reporting

I'll look into it ASAP

**Comment by dpgaspar** [comment]
related with #1618

**Comment by dpgaspar** [comment]
@shawnzhu would love to ear your thoughts on this #1707 possible solution

**Comment by shawnzhu** [comment]
This issue can be fixed by #1707

**Comment by shawnzhu** [comment]
fixed by #1707 and tested by using FAB 3.4.0.

**Comment by usamaB** [comment]
I tried the 1.4.0rc1 and it hasn't been fixed. I'm unable to logout. It just keeps me logged in. And the login flow is incorrect as well. If you have a single provider it doesn't show login UI but takes directly to enter email/pass UI(google in my case).

**Comment by shawnzhu** [comment]
@usamaB may I ask if it installs FAB 3.4.0 when you installs superset 1.4.0rc1? the ☝️ sounds like it still uses FAB 3.3.4.

you can do `pip list | grep flask-appbuilder` to see the version number of installed FAB.

**Comment by usamaB** [comment]
it's running 3.3.0 actually.  I'm using the helm chart.

**Comment by shawnzhu** [comment]
I found it still packages FAB 3.3.4 in superset 1.4.0rc1 so this should be surfaced to superset community:

```SHELL
> docker run -it --rm apache/superset:1.4.0rc1 /bin/bash -c "pip3 list | grep Flask-AppBuilder" 
Flask-AppBuilder       3.3.4
```

**Comment by usamaB** [comment]
I did it by going into the Worker and the superset pod shell.

**Comment by shawnzhu** [comment]
the fix of this issue has been included in the main branch of superset by https://github.com/apache/superset/pull/17420 which is *NOT* included in superset 1.4.0rc1 (just [cancelled](https://lists.apache.org/thread/q4r5y0xr84xsk6rbq50wx9zjv0vcchn2)) yet. see https://github.com/apache/superset/blob/361e510cae4eb9eb7e7bee095ef419a046e00cdf/setup.py#L78

So I would suggest when a new RC release comes out for vote in [superset maillist](https://lists.apache.org/list.html?dev@superset.apache.org)
